client/helpers.go

```diff
@@ -377,3 +385,47 @@ func TagsToMap(tagSlice interface{}) map[string]string {
 	TagsIntoMap(tagSlice, ret)
 	return ret
 }
+
+func ListAndDetailResolver(ctx context.Context, meta schema.ClientMeta, res chan<- interface{}, list ListResolverFunc, details DetailResolverFunc) error {
+	var diags diag.Diagnostics
+
+	errorChan := make(chan error)
+	detailChan := make(chan interface{})
+	// Channel that will communicate with goroutine that is aggregating the errors
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		for detailError := range errorChan {
+			diags = diags.Add(diag.FromError(detailError, diag.RESOLVING))
+		}
+	}()
+	sem := semaphore.NewWeighted(int64(MAX_GOROUTINES))
+
+	go func() {
+		defer close(errorChan)
+		for item := range detailChan {
+			if err := sem.Acquire(ctx, 1); err != nil {
+				continue
+			}
+			func(summary interface{}) {
+				defer sem.Release(1)
+				details(ctx, meta, res, errorChan, summary)
+			}(item)
+		}
+	}()
+
+	err := list(ctx, meta, detailChan)
+	close(detailChan)
+	if err != nil {
+		return diag.WrapError(err)
+	}
+
+	// All items will be attempted to be fetched, and all errors will be aggregated
+	<-done
+
+	// Only return the diags if there is an error
+	if diags.HasErrors() {
```

Comment:
```suggestion
	if diags.HasDiags() {
```
HasErrors implies ERROR severity but it could also return warnings